site/src/pages/DeploySettingsPage/OAuth2AppsSettingsPage/EditOAuth2AppPageView.tsx

```diff
@@ -0,0 +1,300 @@
+  mutatingResource: MutatingResource;
+  updateApp: (req: TypesGen.PutOAuth2ProviderAppRequest) => void;
+  deleteApp: (name: string) => void;
+  generateAppSecret: () => void;
+  deleteAppSecret: (id: string) => void;
+  secrets?: readonly TypesGen.OAuth2ProviderAppSecret[];
+  fullNewSecret?: TypesGen.OAuth2ProviderAppSecretFull;
+  ackFullNewSecret: () => void;
+  error?: unknown;
+};
+
+export const EditOAuth2AppPageView: FC<EditOAuth2AppProps> = ({
+  app,
+  isLoadingApp,
+  isLoadingSecrets,
+  mutatingResource,
+  updateApp,
+  deleteApp,
+  generateAppSecret,
+  deleteAppSecret,
+  secrets,
+  fullNewSecret,
+  ackFullNewSecret,
+  error,
+}) => {
+  const theme = useTheme();
+  const [searchParams] = useSearchParams();
+  const [showDelete, setShowDelete] = useState<boolean>(false);
+
+  return (
+    <>
+      <Stack
+        alignItems="baseline"
+        direction="row"
+        justifyContent="space-between"
+      >
+        <Header
+          title="Edit OAuth2 application"
+          description="Configure an application to use Coder as an OAuth2 provider."
+        />
+        <Button
+          component={Link}
+          startIcon={<KeyboardArrowLeft />}
+          to="/deployment/oauth2-provider/apps"
+        >
+          All OAuth2 Applications
+        </Button>
+      </Stack>
+
+      {fullNewSecret && (
+        <ConfirmDialog
+          hideCancel
+          open={Boolean(fullNewSecret)}
+          onConfirm={ackFullNewSecret}
+          onClose={ackFullNewSecret}
+          title="OAuth2 client secret"
+          confirmText="OK"
+          description={
+            <>
+              <p>
+                Your new client secret is displayed below. Make sure to copy it
+                now; you will not be able to see it again.
+              </p>
+              <CodeExample
+                code={fullNewSecret.client_secret_full}
+                css={{
+                  minHeight: "auto",
+                  userSelect: "all",
+                  width: "100%",
+                  marginTop: 24,
+                }}
+              />
+            </>
+          }
+        />
+      )}
+
+      <Stack>
+        {searchParams.has("created") && (
+          <Alert severity="info" dismissible>
+            Your OAuth2 application has been created. Generate a client secret
+            below to start using your application.
+          </Alert>
+        )}
+
+        {error ? <ErrorAlert error={error} /> : undefined}
+
+        {isLoadingApp && <Loader />}
+
+        {!isLoadingApp && app && (
+          <>
+            <DeleteDialog
+              isOpen={showDelete}
+              confirmLoading={mutatingResource.deleteApp}
+              name={app.name}
+              entity="OAuth2 application"
+              onConfirm={() => deleteApp(app.name)}
+              onCancel={() => setShowDelete(false)}
+            />
+
+            <Stack direction="row">
+              <div>
+                <strong>Client ID:</strong>
```

Comment:
I know this was done to make the text bold, but for accessibility/semantics, I'm wondering if it's be better to mark this up as a header, and then style it the same as before

`<strong>` is generally for things that have urgency/seriousness, like warnings

Reply:
Good point!  Will fix.